sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRelation.scala

```diff
@@ -78,7 +78,8 @@ private[sql] object JDBCRelation extends Logging {
     // Overflow and silliness can happen if you subtract then divide.
     // Here we get a little roundoff, but that's (hopefully) OK.
     val stride: Long = upperBound / numPartitions - lowerBound / numPartitions
-    val column = partitioning.column
+    val dialect = JdbcDialects.get(jdbcOptions.url)
+    val column = dialect.quoteIdentifier(partitioning.column)
```

Comment:
We should do it in `class JDBCOptions`. To avoid breaking the behavior, we should eat the quotes if users manually specify them.

Reply:
We also need to add a test case in `PostgresIntegrationSuite`